### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by Alesia-V175**
1.Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module03
2. Screenshot :
image
image
image

image

3. Deploy: https://alesia-v157-react-rsschool.netlify.app/
4. Done 01.04.2023 / Deadline 03.04.2023
5. Score: 15 / 15

There is you can find full PR to main brach: 
https://github.com/Alesia-V175/react-rs-school/pull/31

Instructions:

You need to clone this repo and checkout the branch 'hooks' branch or 'develop' branch.

1. npm install (install required packages)
2. npm run dev (start project in developer mode)
3. npm run lint (check code style)
4. npm run build (prepare application build)
5. npm test (run tests)

If you have any questions, please connect with me in Discord #Alesia (Alesia-V175)#6265

### CODE REVIEW

File: src/components/UI/FormInputFile/FormInputFile.test.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+interface Props {
+  type: InputType,
+  name?: string,
+  title?: string,
+  error?: FieldError,
+  save: (file: string) => void,
+}
+describe('Component: FormSelect', () => {
+  it('should render error', () => {
```
Comment (Ksarelto) [inline src/components/UI/FormInputFile/FormInputFile.test.tsx:16]: u can use just one describe for one component

**Review by Ksarelto — COMMENTED** [review]
many similar mistakes, please, check all files for them

File: src/App/App.test.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { describe, it } from 'vitest';
+import { screen, render } from '@testing-library/react';
+import App from './App';
+
+describe('Component: Footer', () => {
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/App/App.test.tsx:7]: fixed

File: src/components/FormCardItem/FormCardItem.tsx [hunk 3]
```diff
@@ -1,20 +1,27 @@
-import { IFormCardItem } from '../../types/interfaces';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/FormCardItem/FormCardItem.tsx:1]: fixed

File: src/components/FormCardItem/FormCardItem.tsx [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
-    <div><span>Name: </span>{name}</div>
-    <div><span>E-mail: </span>{email}</div>
-    <div><span>Date: </span>{date}</div>
-    <div><span>Payment: </span>{payment}</div>
-    <div><span>Type of photo shoot: </span>{type}</div>
-    <div><span>Confirmation: </span>{confirm}</div>
-  </li>
-);
+const FormCardItem = (data: IFormCard) => {
+  const date = new Date(data.date).toISOString().split('T')[0];
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/FormCardItem/FormCardItem.tsx:6]: fixed

File: src/components/FormCardItem/FormCardItem.tsx [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
+        <img src={data.file} alt={data.name} className={styles.card__img}/>
+      </div>
+      <div><span>Name: </span>{data.name}</div>
+      <div><span>E-mail: </span>{data.email}</div>
+      <div><span>Date: </span>{date}</div>
+      <div><span>Payment: </span>{data.radio}</div>
+      <div><span>Type of photo shoot: </span>{data.type}</div>
+      <div><span>Confirmation: </span>
+        {data.checkbox && (
+          <>Yes</>
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/FormCardItem/FormCardItem.tsx:20]: fixed

File: src/components/Header/Header.test.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { describe, it } from 'vitest';
+import { screen, render } from '@testing-library/react';
+import Header from './Header';
+
+describe('Component: Footer', () => {
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/Header/Header.test.tsx:7]: fixed

File: src/components/Header/Header.tsx [hunk 7]
```diff
@@ -13,48 +13,44 @@ const getRouteTitle = () => {
...
-          </nav>
-        </div>
-      </header>
-    );
-  }
-}
+          <NavLink to="/about"
+                   className={setActiveLink}
+                   data-testid="about-link"
+                   onClick={() => updateHeaderTitle(Paths.about.title)}>About us
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/Header/Header.tsx:43]: updated

File: src/components/Header/Header.tsx [hunk 8]
```diff
@@ -13,48 +13,44 @@ const getRouteTitle = () => {
...
+          </NavLink>
+          <NavLink to="/form"
+                   className={setActiveLink}
+                   data-testid="form-link"
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/Header/Header.tsx:47]: fixed

File: src/components/OrderForm/OrderForm.test.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { vitest } from 'vitest';
+import OrderForm from './OrderForm';
+
+const getFormItems = vitest.fn();
+describe('Component: OrderForm', () => {
+  it('should render form', () => {
+    render(<OrderForm setItems={getFormItems} />);
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/OrderForm/OrderForm.test.tsx:9]: added the construction before tests

File: src/utils/checkDate.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+const isDateInFuture = (date: Date) => {
+  const todayDate = new Date();
+  const userDate = new Date(date);
+
+  return userDate.getTime() > todayDate.getTime();
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/utils/checkDate.ts:5]: fixed

File: src/pages/NotFound/NotFound.tsx [hunk 11]
```diff
@@ -2,15 +2,17 @@ import { Link } from 'react-router-dom';
...
 
-const NotFound = () => (
-  <>
-    <div className={styles.container}>
-      <img src={imageNotFound} alt="page not found" className={styles.container__image} />
-      <div className={styles.container__button}>
-       <Link to="/" className={styles.container__link}>Go home</Link>
+const NotFound = (): JSX.Element => {
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/pages/NotFound/NotFound.tsx:7]: fixed

File: src/pages/Home/Home.tsx [hunk 12]
```diff
@@ -1,37 +1,27 @@
...
+import Search from '../../components/Search/Search';
 import banner from '../../assets/images/banner-main.jpg';
 import styles from './Home.module.scss';
 
-class Home extends React.Component<object, { cards: ICardItem[] }> {
-  state = {
-    cards: [],
-  };
+const Home = (): JSX.Element => {
+  const [cards] = useState(cardsDB);
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/pages/Home/Home.tsx:9]: fixed

File: src/pages/Home/Home.test.tsx [hunk 13]
```diff
@@ -1,9 +1,9 @@
 import { describe, it } from 'vitest';
 import { render, screen } from '@testing-library/react';
-import Home from './Home.js';
+import Home from './Home';
 
-describe('Header', () => {
-  it('Home page should have a banner ', () => {
+describe('Component: Header', () => {
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/pages/Home/Home.test.tsx:5]: fixed

File: src/components/UI/FormInputFile/FormInputFile.tsx [hunk 14]
```diff
@@ -1,35 +1,39 @@
 import React, { forwardRef, ReactNode } from 'react';
+import { FieldError } from 'react-hook-form';
 import { InputType, Ref } from '../../../types/types';
 import styles from './FormInputFile.module.scss';
 
 interface Props {
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/UI/FormInputFile/FormInputFile.tsx:6]: update name of interfaces

File: src/components/UI/FormInputFile/FormInputFile.tsx [hunk 15]
```diff
@@ -1,35 +1,39 @@
...
+      title={props.title}
+      onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+        if (event.currentTarget.files) {
+          props.save(URL.createObjectURL(event.currentTarget.files[0]));
+        }
+      }}
+    />
+    <p className={styles.error__wrap}>
+      {props.error && props.error.message && (
+        <>{props.error.message}</>
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/UI/FormInputFile/FormInputFile.tsx:33]: fixed

File: src/components/UI/FormInputFile/FormInputFile.test.tsx (hunk 1 above)
Comment (Alesia-V175) [reply src/components/UI/FormInputFile/FormInputFile.test.tsx:16]: fixed

File: src/components/OrderForm/OrderForm.test.tsx [hunk 16]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { vitest } from 'vitest';
+import OrderForm from './OrderForm';
+
+const getFormItems = vitest.fn();
+describe('Component: OrderForm', () => {
```
[the remark this answers is not in the document]
Comment (Alesia-V175) [reply src/components/OrderForm/OrderForm.test.tsx:7]: fixed

**Pull request merged**